### PR TITLE
1154: handle the links in the footer

### PR DESCRIPTION
**Pull request by szidi-eross**
Task: [#1154](https://galaxis.planio.com/issues/1154?issue_count=15&issue_position=1&next_issue_id=1131)

### CODE REVIEW

File: src/galaxis-footer/pages/pages.component.tsx [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
 
 type PagesProps = { pages: PageModel[] };
 
-const Pages: FC<PagesProps> = ({ pages }) => {
+const Pages: FC<PagesProps> = ({ pages: initialPages }) => {
+  const [pages, setPages] = useState<PageModel[]>(initialPages);
+
+  useEffect(() => {
+    const path = window.location.pathname;
+    if (path.includes("engines")) {
```
Comment (levikovendi) [inline src/galaxis-footer/pages/pages.component.tsx:14]: The `engines` is an example in the ticket. We discussed to many of Apps will use this component.

File: src/galaxis-footer/pages/pages.component.tsx [hunk 2]
```diff
@@ -1,19 +1,38 @@
...
+  }, [initialPages]);
+
   return (
     <StyledWrapper>
       <CustomTitle className="title" title="Pages" />
       <Box className="pages--box" data-testid="pages-box">
         {pages.map((page, index) => (
           <Typography className="pages--typography" key={index} variant="main">
-            <a>{page.label}</a>
+            <Link
```
Comment (levikovendi) [inline src/galaxis-footer/pages/pages.component.tsx:29]: I don't see where we handle the link. If we are in the Galaxis website the link should be for example "/privacy-policy" in any other place the link should be "https://galaxis.xyz/privacy-policy".

**Comment by levikovendi** [comment]
Please create a handler for Contact us too.

File: src/utils/links.util.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+function generateContactFormUrl(url: string, subjectForContextUS: string): string | null {
+  if (containsContactForm(url)) {
+    const encodedSubject = encodeURIComponent(subjectForContextUS);
+    return `https://galaxis.xyz?subject=${encodedSubject}`;
+  }
+  return null;
+}
+
+function extractSubjectFromUrl(url: string): string | null {
```
Comment (levikovendi) [inline src/utils/links.util.ts:40]: Please use the "const" version instead of it

File: src/utils/links.util.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  }
+  return false;
+}
+
+function removeDomainFromUrl(url: string): string {
+  const pathRegex = /^(?:https?:\/\/)?[^\/]*(\/.*)/;
+  const match = url.match(pathRegex);
+
+  if (match && match.length > 1) {
+    return match[1];
```
Comment (levikovendi) [inline src/utils/links.util.ts:22]: 1???

File: src/utils/links.util.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  }
+  return url;
+}
+
+function containsContactForm(url: string): boolean {
+  const regex = /contact-form/i;
+  return regex.test(url);
+}
+
+function generateContactFormUrl(url: string, subjectForContextUS: string): string | null {
```
Comment (levikovendi) [inline src/utils/links.util.ts:32]: subjectForContextUS -> subjectForContactUs

File: src/galaxis-footer/contact-us/contact-us.component.tsx [hunk 6]
```diff
@@ -1,23 +1,64 @@
...
+const ContactUs: FC<Props> = ({ socialMediaIcons: initialIcons, address }) => {
+  const [socialMediaIcons, setSocialMediaIcons] = useState<SocialMediaIcon[]>(initialIcons);
+
+  const iconsMapper = useCallback(
+    (currentHostName: string): SocialMediaIcon[] => {
+      return socialMediaIcons.map((item) => {
+        const currentUrl = getCurrentDomain(item.url);
+        const sameHost = areUrlsSame(currentHostName, currentUrl);
+
+        const subjectForContextUS = "custom subject here";
```
Comment (levikovendi) [inline src/galaxis-footer/contact-us/contact-us.component.tsx:29]: subjectForContactUs

File: src/utils/links.util.ts [hunk 7]
```diff
@@ -0,0 +1,61 @@
+const getCurrentDomain = (url: string): string => {
```
[the remark this answers is not in the document]
Comment (szidi-eross) [reply src/utils/links.util.ts:4]: handled it in the domainRegex

**Review by levikovendi — APPROVED** [review]

**Pull request merged**